content/blog/2019/12/2019-12-06-introducing-aws-secrets-manager-credentials-provider-plugin.adoc

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title: Introducing the AWS Secrets Manager Credentials Provider for Jenkins
+tags:
+- pipeline
+- plugins
+author: chriskilding
+---
+
+API keys and CI secrets are difficult to handle safely, and probably something you avoid thinking about. In this post I'll show how the new https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin[AWS Secrets Manager Credentials Provider] plugin allows you to marshal your secrets into one place, and use them securely from Jenkins.
+
+When CI pipelines moved to the public cloud, credential management did not evolve with them. If you're in this situation, you may have seen a number of tactical workarounds to keep Jenkins builds talking to the services they depend on. The workarounds range from bad (hardcoding plaintext secrets into Git) to merely painful (wrangling Hiera EYAML), but their common feature is that they tend to make copies of secrets beyond the reach of automation. This increases their attack surface, makes routine key rotation impractical, and makes remediation difficult after a breach.
+
+The good news is that there is a better way!
+
+AWS Secrets Manager is a comprehensive solution for secure secret storage. You define a secret just once for your whole AWS account, then you give your consumers permission to use the secrets. Secrets Manager lets you manage a secret entry (name and metadata) separately from its value, and it integrates with other AWS services that you already use:
+
+- Secret entry management: Manual (Web console, AWS CLI) or with an infrastructure management tool (https://www.terraform.io/docs/providers/aws/r/secretsmanager_secret.html[Terraform], CloudFormation etc.)
+- Secret value management: Manual (Web console, AWS CLI) or automatic (secret rotation Lambda function).
+- Access control: AWS IAM policies (for both applications and human operators).
+- Secret encryption: Amazon KMS automatically encrypts the secret value. Use either the account's default KMS key, or a customer-managed KMS key.
+- Auditing: AWS CloudTrail and CloudWatch Events.
+
+A couple of teams in my company started to use Secrets Manager from Jenkins jobs by calling the AWS CLI, but this remained a niche approach as it was quite unwieldy. There was clearly an appetite to integrate key developer apps with a centralised secrets store, but production-ready integrations were needed for wider adoption. So this year I created the https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin[AWS Secrets Manager Credentials Provider] plugin for Jenkins, with help from friends in the Jenkins community, to do exactly that.
```

Comment:
> in my company

This reference indicates to me that a more complete bio would make sense to tell readers more about who you are.

Reply:
I've added my LinkedIn URL to the bio to tell readers more.